Maxis/Infrastructure/Repositories/UserRepository.cs

```diff
@@ -1,168 +1,184 @@
             }
+
         }
 
-        /// <summary>
-        /// 
-        /// </summary>
-        /// <param name="model"></param>
-        public void Update(EditUserViewModel editViewModel)
+        public bool Update(EditUserViewModel editViewModel)
         {
-           
             try
             {
-                    var entity = _db.ONNET_USER.FirstOrDefault(u => u.UserId == editViewModel.UserId);
-                    if (entity != null)
-                    {
-                        entity.UserId = editViewModel.UserId;
-                        entity.Username = editViewModel.Username;
-                        entity.Email = editViewModel.Email;
-                        entity.Mobile = editViewModel.Mobile;
-                        entity.Department = editViewModel.Department;
-                        entity.Title = editViewModel.Title;
-                        entity.Status = editViewModel.Status;
-                    }
-                    _db.SaveChanges();
+                var entity = _db.ONNET_USER.FirstOrDefault(u => u.UserId == editViewModel.UserId);
+                if (entity != null)
+                {
+                    entity.UserId = editViewModel.UserId;
+                    entity.Email = editViewModel.Email;
+                    entity.Mobile = editViewModel.Mobile;
+                    entity.Department = editViewModel.Department;
+                    entity.Title = editViewModel.Title;
+                    entity.Status = editViewModel.Status;
+                    entity.RoleId = editViewModel.RoleId;
+                }
+
+                _db.SaveChanges();
+                return true;
             }
-            catch (Exception)
+            catch (Exception ex)
             {
-                throw;
+                throw ex;
+            }
+        }
+
+        public UserDetailsViewModel SelectByUser(string username)
+        {
+            var user = (from ep in _db.ONNET_USER
+                        join e in _db.ONNET_USERROLE on ep.RoleId equals e.RoleId
+                        where ep.Username == username
+                        select new UserDetailsViewModel()
+                        {
+                            UserId = ep.UserId,
+                            Username = ep.Username,
+                            Email = ep.Email,
+                            Mobile = ep.Mobile,
+                            Department = ep.Department,
+                            Title = ep.Title,
+                            Status = ep.Status,
+                            Roles = e.RoleName
+                        }).Single();
+            return user;
+        }
+
+        public static string Encrypt(string password)
+        {
+            password = password ?? "";
+            var encoding = new System.Text.ASCIIEncoding();
+            var res = "";
+            var keyByte = encoding.GetBytes(password);
+            var messageBytes = encoding.GetBytes(ConfigurationManager.AppSettings["Hashkey"]);
```

Comment:
salt key should not be same for all user, new key should be generated for each user.